### DISCUSSION TITLE
Rename the project, licensing issue with name "mxGraph"

### DISCUSSION THREAD

**Discussion by coclav** [maxGraph/maxGraph#23] [Development]
Hello team, 

I have followed up with the creators of the original creator of the "mxGraph" library off which this project is based. 

Long story short, we **cannot use the name mxGraph**. 

We need to find a new project name. I have put a couple of suggestions in the replies, please "upvote" for your favourite or propose some new ones.

I would suggest that we review progress on may 1st 2021, and go ahead with the one with the most "upvotes".

cc @jgadsden @junsikshim @mcyph @tbouffard

**Comment by coclav** [comment]
### maxGraph

distinctive yet familiar.

**Reply by mcyph** [reply]
Edging towards this one because I could remember this one since yesterday but had to think a few seconds before remembering my own!

**Comment by coclav** [comment]
### freeGraph
(or freedomGraph) 

Strengthen the message that it's about creating a free-forever library; and also that this is a very powerful library allowing to do pretty much anything

**Comment by mcyph** [comment]
Not an answer for voting but think it would be nice if it was short and pronounceable. One reason for this is HTML components need a hyphen character, and shorter could mean less characters if using it as a prefix, e.g. a circle could be `<my-circle>` or `<nxt-circle>` pronounced "next-circle" (if we want to add HTML components, that is). I think I read in the book "code complete" or similar that words that are pronounceable take less time to process (I'm having trouble finding the citation and don't have it with me though).

**Comment by mcyph** [comment]
### veGraph 

Short and has the letters "vertices and edges", also getting to the point in defining to new users what the lib does.

**Comment by jgadsden** [comment]
It does not matter to the Threat Dragon project what it is called, because we are just starting out to use mxgraph.

But mxgraph appears in many (probably thousands) project's package.json, so if it is renamed then (if I understand correctly) their project will not get updates from npmjs.com until this mxgraph package is renamed in their package.json . This is not a great thing to ask the mxgraph community to do

@coclav what was their legal reasoning? The community has been using the name 'mxgraph' and the source code for some time now under Apache 2.0 licence. Is the problem that the name is copyright? Or there are claims on the source code?

**Reply by lreading** [reply]
If this is renamed, we could reach out to the original authors to see if they would be kind enough to add a depreciated notice to the npm package, pointing users to this repo/package.

I assume with a new name, we would start pushing a new NPM package?  Are we close to deciding on a name yet?  I'm beginning work on the Threat Dragon migration, and am hesitant to add an unmaintained package.  I'd be happy to help get work going on the renaming process and creating a new package!

CC: @coclav

**Comment by junsikshim** [comment]
We could also have some generic name like 'draw-graph.js'.
It's clear what the library is about, and it pays homage to the old draw.io ;)

**Comment by jgadsden** [comment]
@coclav what name do you think is best? It would be great to decide on this soon, as it is now after May 1st.

The most votes (by a long way) went to 'maxGraph' - is this OK to @junsikshim @mcyph @tbouffard @lreading 

I can create an npm package and make this available on npmjs.com (this is a well trod path, and we are used to doing it for OWASP Threat Dragon core) - but it has to be named so that it can start being referenced as a dependency.

@lreading is planning the work to get mxgraph-as-was into Threat Dragon, so this is coming up soon for us

Note the discussion in a recent PR: https://github.com/jsGraph/mxgraph/pull/33#issuecomment-834399313

**Comment by jgadsden** [comment]
Just to make clear what @coclav is saying in this discussion, as this question has come up in a pull-request.

@tbouffard has raised the point about the licence before and after mxgraph 4.2.2 release. At this point the licence deviated from standard Apache Licence version 2.0 on October 29th 2020.

Colin has had a discussion with the original mxgraph owners / contributors and they have confirmed on 16th April:

"
_So I had the confirmation from the guys at mxGraph that we cannot use the name mxGraph, apart from that we are all good. I posted a message about it there : https://github.com/jsGraph/mxgraph/discussions/23_ 
"

**Comment by Kurt29** [comment]
Hm, why not something like tsGraph or jsGraph instead of mxgraph?

It is simple, short and I think it says enough about the project without limiting its scope

**Reply by csouchet** [reply]
**jsGraph** is a little confusing. 

There is some librairies which use **jsGraph** in their names:
- http://js-graph-it.sourceforge.net/
- https://www.npmjs.com/package/node-jsgraph
- https://www.js-graph.com/
- https://www.ndss-symposium.org/wp-content/uploads/2018/02/ndss2018_07B-4_Li_paper.pdf

**Reply by csouchet** [reply]
**tsGraph** already exists. It is a fork of **mxGraph** library: https://www.npmjs.com/package/tsgraph
